### PR TITLE
Refactor parsing, start blend mode support, improve attribute inheritence/processing

### PR DESCRIPTION
**Pull request by dnfield**
Tests are passing, with this change some more of the images in the internal vgtest repo are now rendering identically to flutter_svg.

This still needs some more tests I think. There's also a couple things weird around use that I _don't think_ are regressions but still aren't rendering the same as flutter_svg.

### CODE REVIEW

**Comment by dnfield** [comment]
this makes most out of order references work, except for gradients, which are too eagerly resolving colors/stops right now. That should be fixable but this patch is already pretty bug.

File: packages/vector_graphics_compiler/lib/src/svg/node.dart [hunk 1]
```diff
@@ -145,96 +114,101 @@ class ParentNode extends PaintingNode {
...
     ).._children.addAll(_children);
   }
 
-  // Whether or not a save layer should be inserted at this node.
-  bool _requiresSaveLayer() {
-    final Paint? localPaint = paint;
-    if (localPaint == null) {
-      return false;
+  Paint? _createLayerPaint() {
+    if (attributes.blendMode != null) {
```
Comment (jonahwilliams) [inline packages/vector_graphics_compiler/lib/src/svg/node.dart:159]: So how do we determine whether or not we need a layer - just if there is a blend mode?

Comment (dnfield) [reply packages/vector_graphics_compiler/lib/src/svg/node.dart:159]: There are other cases like masks and clips applied to multiple children. I think I'm missing something about group opacity but my test case for that seems to be passing...

Comment (dnfield) [reply packages/vector_graphics_compiler/lib/src/svg/node.dart:159]: My test case is _not_ passing lol.

Comment (dnfield) [reply packages/vector_graphics_compiler/lib/src/svg/node.dart]: Fixed and updated/added test

**Review by jonahwilliams — APPROVED** [review]
LGTM!

**Comment by dnfield** [comment]
This broke some types of gradients. Looking into that.

**Pull request merged**